dot/state/service.go

```diff
@@ -175,9 +192,27 @@ func (s *Service) Start() error {
 
 	// Start background goroutine to GC pruned keys.
 	go s.Storage.pruneStorage(s.closeCh)
+
+	if s.Metrics.Publish {
+		go s.updateMetrics()
+	}
 	return nil
 }
 
+func (s *Service) updateMetrics() {
+	ticker := time.NewTicker(s.Metrics.Interval)
+	for {
+		select {
+		case <-s.closeCh:
+			return
+		case <-ticker.C:
+			transactionPoolGauge.Set(float64(s.Transaction.pool.Len()))
+			transactionQueueGauge.Set(float64(s.Transaction.queue.Len()))
+			leavesGauge.Set(float64(len(s.Block.Leaves())))
```

Comment:
Shouldn't this be set by what is dealing with these directly? I would rather dependency inject a metrics interface where it's needed that have it set values with a mutex with these ticker based goroutines.

Reply:
I was migrating how it was originally done.  I'll instrument the sub types.